controller/rancher/certificate_fetcher.go

```diff
@@ -286,19 +287,22 @@ func (fetcher *RCertificateFetcher) readCertificate(path string, f os.FileInfo,
 			if !file.IsDir() {
 				contentBytes, err := fetcher.evaluatueLinkAndReadFile(path, file.Name())
 				if err != nil {
-					return err
-				}
-				if file.Name() == fetcher.CertName {
-					isCertFound = true
-					cert.Cert = string(*contentBytes)
-				} else if file.Name() == fetcher.KeyName {
-					isCertFound = true
-					cert.Key = string(*contentBytes)
+					logrus.Errorf("Error while reading file [%v]. Error: %v", file.Name(), err)
+				} else {
+					if file.Name() == fetcher.CertName {
+						isCertFound = true
+						cert.Cert = string(*contentBytes)
+					} else if file.Name() == fetcher.KeyName {
+						isKeyFound = true
+						cert.Key = string(*contentBytes)
+					}
 				}
 			}
 		}
-		if isCertFound {
+		if isCertFound && isKeyFound {
 			fetcher.tempCertsMap[path] = &cert
+		} else if isCertFound || isKeyFound {
+			logrus.Infof("Skipping incomplete cert found under dir [%v], [isCertFound %v] [isKeyFound %v]", path, isCertFound, isKeyFound)
```

Comment:
I believe that will be a non-error state, since it just means there is a subdirectory without any cert file in it. That is possible, we should just ignore such a directory, no need to raise error.